tests/Kiota.Builder.Tests/KiotaBuilderTests.cs

```diff
@@ -1434,8 +1434,8 @@ public void Inline_Property_Inheritance_Is_Supported()
         var resourceClass = codeModel.FindNamespaceByName("ApiSdk.models").FindChildByName<CodeClass>("resource");
         var itemsNS = codeModel.FindNamespaceByName("ApiSdk.resource.item");
         var responseClass = itemsNS.FindChildByName<CodeClass>("ResourceResponse");
-        var derivedResourceClass = itemsNS.FindChildByName<CodeClass>("ResourceResponse_derivedResource");
-        var derivedResourceInfoClass = itemsNS.FindChildByName<CodeClass>("ResourceResponse_derivedResource_info");
+        var derivedResourceClass = itemsNS.FindChildByName<CodeClass>("resource");
```

Comment:
I'm not sure this name change makes a lot of sense:
- resource is the name of the endpoint
- this derived class is one form of all of and should be present (at least we should have derviedResource in the tree)

Reply:
Thanks for pointing this out. 
The class is indeed the same, the issue is that the name has been pulled from `AllOf` (base type). Will need to fix and resolve.

Reply:
Thanks for catching this. Digging into this, I now have a better understanding of the `GetSchemaNames`/`FlattenIfRequired` is intended to do here. And the fix shouldn't have resided there. 
Resolved in latest update.